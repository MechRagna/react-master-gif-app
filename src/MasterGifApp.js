// rafc --> snippet para crear componentes funcionales
import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
import PropTypes from "prop-types";

export const MasterGifApp = ({ defaultCategories }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Master Gif</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

MasterGifApp.propTypes = {
  defaultCategories: PropTypes.array,
};

MasterGifApp.defaultProps = {
  defaultCategories: [],
};
