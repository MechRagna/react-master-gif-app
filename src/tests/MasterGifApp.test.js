import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { MasterGifApp } from "../MasterGifApp";

describe("Tests en <MasterGidApp />", () => {
  test("Debe matchear con el snapshot", () => {
    const wrapper = shallow(<MasterGifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar la lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<MasterGifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
