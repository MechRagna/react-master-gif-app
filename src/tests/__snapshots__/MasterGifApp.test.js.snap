// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en <MasterGidApp /> Debe matchear con el snapshot 1`] = `
<Fragment>
  <h2>
    Master Gif
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests en <MasterGidApp /> Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h2>
    Master Gif
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
